### ISSUE TITLE
Environment Variable Init: HOME

### ISSUE THREAD

**Issue by mitnk** [containers/youki#680]
Using youki as the runtime will missing the env HOME:
```
$ docker run --rm --runtime youki busybox env
HOSTNAME=efff650756b1
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
```

**Expected** (as using runc):
```
$ docker run --rm busybox env
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
HOSTNAME=2ec2c9640467
HOME=/root
```

Where is environment variables initialized? I tried but not found in code of the repo.

**Comment by utam0k** [comment]
@mitnk Thanks for your report. We set the environment variables here.
https://github.com/containers/youki/blob/a8e4003bc67b0eef7fcd59ec89928f7436d5c439/crates/libcontainer/src/process/container_init_process.rs#L415-L419

I couldn't find where the HOME environment variable gets, and I couldn't find it in the runtime-spec.
https://github.com/opencontainers/runtime-spec/blob/main/config.md

I'll do some more research. Do you know anything about it?

**Comment by utam0k** [comment]
I found that runc has been implemented here. 
https://github.com/opencontainers/runc/blob/eddf35e5462e2a9f24d8279874a84cfc8b8453c2/libcontainer/init_linux.go#L309-L398

**Comment by utam0k** [comment]
Hi, @mitnk. If you have interested in this implementation, I'll leave this implementation to you. Of course, if you don't have time to do it, feel free to ask me to implement it 🙏

**Comment by mitnk** [comment]
Thanks for the info, I'll have a try :)

**Comment by utam0k** [comment]
> Thanks for the info, I'll have a try :)

Awesome! If you have any questions, please do not hesitate to contact me.

**Comment by mitnk** [comment]
@utam0k Question 1: I tried adding a `syscall.get_pwdir(proc.user().uid())` before [setting envs](https://github.com/containers/youki/blob/a8e4003bc67b0eef7fcd59ec89928f7436d5c439/crates/libcontainer/src/process/container_init_process.rs#L415-L419), but the file `passwd` seems empty (all fields are null-like). So I cannot get the home of user like this. Any ideas?

**Comment by mitnk** [comment]
Draft PR: #681

**Comment by mitnk** [comment]
Another question: how to see the logs printed by code like `xxx.with_context(|| "some msg")`, while using `docker run --runtime youki ...`? Do we have a doc on this anywhere (I haven't read all the docs yet).

**Comment by mitnk** [comment]
The file `/etc/passwd` exists. It's `libc::getpwuid_r()` returning `2` (No such file/dir?). still checking the reason.

Content of `/etc/passwd` (via `fs::read_to_string()` after `chroot()`):
```
root:x:0:0:root:/root:/bin/sh
daemon:x:1:1:daemon:/usr/sbin:/bin/false
bin:x:2:2:bin:/bin:/bin/false
sys:x:3:3:sys:/dev:/bin/false
sync:x:4:100:sync:/bin:/bin/sync
mail:x:8:8:mail:/var/spool/mail:/bin/false
www-data:x:33:33:www-data:/var/www:/bin/false
operator:x:37:37:Operator:/var:/bin/false
nobody:x:65534:65534:nobody:/home:/bin/false
```

**Comment by mitnk** [comment]
Not sure why, but a re-build fixed the issue. It just works now..

**Comment by utam0k** [comment]
> Another question: how to see the logs printed by code like `xxx.with_context(|| "some msg")`, while using `docker run --runtime youki ...`? Do we have a doc on this anywhere (I haven't read all the docs yet).

It is hard to see the log while using docker. If you want to see the log, it would be good to use the youki directory without using docker.

**Comment by utam0k** [comment]
> @utam0k Question 1: I tried adding a `syscall.get_pwdir(proc.user().uid())` before [setting envs](https://github.com/containers/youki/blob/a8e4003bc67b0eef7fcd59ec89928f7436d5c439/crates/libcontainer/src/process/container_init_process.rs#L415-L419), but the file `passwd` seems empty (all fields are null-like). So I cannot get the home of user like this. Any ideas?

Did you resolve this question by yourself?

**Comment by mitnk** [comment]
Yes, resolved.

**Comment by mitnk** [comment]
fixed in #681 Thanks!

**Comment by utam0k** [comment]
@mitnk Nice work!
